Library/Homebrew/cask/lib/hbc/utils/quarantine.applescript

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/osascript --
+
+use framework "Foundation"
+
+on run(argv)
+  if argv's length < 3 then
+    error "Insufficient parameters: this script needs" ¬
+      & " file_path, data_url, and origin_url"
+  end if
+
+  set filepath to argv's item 1
+  set agent to "Homebrew-Cask"
+  set dataUrl to argv's item 2
+  set originUrl to argv's item 3
+
+  set quarantineProperties to { ¬
+    kLSQuarantineAgentNameKey: agent, ¬
+    kLSQuarantineTypeKey: my kLSQuarantineTypeWebDownload, ¬
+    kLSQuarantineDataURLKey: dataUrl, ¬
+    kLSQuarantineOriginURLKey: originUrl ¬
+  }
+
+  set dataLocationUrl to my NSURL's fileURLWithPath: filepath
+
+  if dataLocationUrl's checkResourceIsReachableAndReturnError: ¬
+    missing value then
+
+    if not dataLocationUrl's setResourceValue: quarantineProperties ¬
+      forKey: my NSURLQuarantinePropertiesKey ¬
+      |error|: missing value then
+
+      error "Homebrew-Cask quarantiner: unable to quarantine " ¬
+        & dataLocationUrl's absoluteString as text
+    end if
+  else
+    error "Homebrew-Cask quarantiner: unable to quarantine " ¬
```

Comment:
Can this use a custom exit code?

Reply:
That’s an annoying limitation of AppleScript. We can choose an error number to be displayed on stderr but we have no influence on the exit code.

Reply:
(My usual workaround for that is to output JSON and have that unwrapped by a shell script wrapper. I don’t feel that would be worth the effort here though.)

Reply:
I've updated the Swift script to just output the error message; then, we could use `command.run(...).stderr.empty?` instead of `success?` to check for errors.

Reply:
Yeh, outputting just the `dataLocationUrl's absoluteString` would be preferable, I think.

Reply:
Updated the script so it only outputs the error message.